libats/src/main/scala/com/advancedtelematic/libats/db/SlickExtensions.scala

```diff
@@ -50,18 +51,38 @@ object SlickExtensions {
   implicit def uuidToJava(refined: Refined[String, Uuid]): Rep[UUID] =
     UUID.fromString(refined.get).bind
 
-  implicit class DbioPaginateExtensions[E, U, A](action: Query[E, U, Seq]) {
+  implicit class DbioPaginateExtensions[E, U](action: Query[E, U, Seq]) {
     def paginateAndSort[T <% slick.lifted.Ordered](fn: E => T, offset: Long, limit: Long): Query[E, U, Seq] = {
       action
         .sortBy(fn)
         .drop(offset)
         .take(limit)
     }
+
     def paginate(offset: Long, limit: Long): Query[E, U, Seq] = {
       action
         .drop(offset)
         .take(limit)
     }
+
+    def paginateAndSortResult[T <% slick.lifted.Ordered](fn: E => T, offset: Long, limit: Long)
+                             (implicit ec: ExecutionContext): DBIO[PaginationResult[U]] = {
+      val tot = action.length.result
+      val pag = action.paginateAndSort(fn, offset, limit).result
+
+      tot.zip(pag).map{ case (total, values) =>
```

Comment:
Why is this zip necessary? Couldn't you just create the `PaginationResult` directly?